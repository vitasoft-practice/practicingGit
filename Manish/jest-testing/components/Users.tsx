import React from 'react'

const Users = ({ user }: any) => {
    return (
        <div>
           <h4 data-testid='userintro' >Hello, THis is {user.name},DOB {user.dob} and email Id {user.email}</h4> 
           
        </div>
        
    )
}

export default Users
