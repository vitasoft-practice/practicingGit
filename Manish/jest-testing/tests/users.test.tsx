import React from 'react'
import { cleanup, render } from '@testing-library/react'
import { create } from 'react-test-renderer'
import Users from '../components/Users'


describe("render users ccomponents correctly", () => {
    afterEach(cleanup);

    const user = {
        name: 'abc',
        dob: '04-06-1999',
        email: 'abc@domain.com'
      
    }
    it('Snapshot matched', () => {
        const userComponent = create(<Users user={user} />);
        expect(userComponent.toJSON()).toMatchSnapshot();

    })

    // it('user props matches', () => {
    //     const { getByTestId } = render(<Users user={user} />);

    //     expect(getByTestId('userintro').textContent).toBe('Hello, THis is abc,DOB 04-06-1999 and email Id abc@domain.com');

    // })
})