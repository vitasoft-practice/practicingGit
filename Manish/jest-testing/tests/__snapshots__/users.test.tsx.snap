// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render users ccomponents correctly Snapshot matched 1`] = `
<div>
  <h4
    data-testid="userintro"
  >
    Hello, THis is 
    abc
    ,DOB 
    04-06-1999
     and email Id 
    abc@domain.com
  </h4>
</div>
`;
